# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocket.h
# Commit: f5353ea9f1df
# Full Hash: f5353ea9f1dfe15a65076bc7aa272d688ebdb8ca
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Add a proxy policy argument to NrSocketProxyConfig, and always set this config on the NrIceCtx. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45101
# ==============================================================================

diff -r ed1ae6a85255 -r f5353ea9f1df media/mtransport/ipc/WebrtcTCPSocket.h
--- a/media/mtransport/ipc/WebrtcTCPSocket.h	Tue Sep 17 18:17:13 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocket.h	Tue Sep 17 18:17:14 2019 +0000
@@ -21,6 +21,7 @@
 #include "mozilla/dom/ipc/IdType.h"     // TabId
 #include "mozilla/dom/PContentChild.h"  // LoadInfoArgs
 #include "nsIProtocolProxyCallback.h"
+#include "mtransport/nr_socket_proxy_config.h"
 
 class nsISocketTransport;
 
@@ -52,7 +53,8 @@
 
   void SetTabId(dom::TabId aTabId);
   nsresult Open(const nsCString& aHost, const int& aPort,
-                const net::LoadInfoArgs& aArgs, const nsCString& aAlpn);
+                const net::LoadInfoArgs& aArgs, const nsCString& aAlpn,
+                NrSocketProxyConfig::ProxyPolicy aProxyPolicy);
   nsresult Write(nsTArray<uint8_t>&& aBytes);
   nsresult Close();
 
@@ -75,6 +77,7 @@
   net::LoadInfoArgs mLoadInfoArgs;
   nsCString mAlpn;
   bool mSsl = false;
+  bool mForceProxy = false;
 
   nsresult DoProxyConfigLookup();
   nsresult OpenWithHttpProxy();